guides/micronaut-dependency-injection-types/micronaut-dependency-injection-types.adoc

```diff
@@ -0,0 +1,55 @@
+common:header-top.adoc[]
+
+== Sample Project
+
+You can link:@sourceDir@.zip[download a sample application] with the code examples shown in this article.
+
+== Introduction
+
+What is a dependency? Class A has a dependency on Class B when it interacts with it in any way. For example, Class A executes a method in an instance of Class B. You typically don't instantiate class dependencies when using Micronaut Framework's dependency injection engine. Instead, you rely on the framework to provide dependency instances.
+
+NOTE: In Micronaut Framework documentation, we often use the term `Bean` and `Dependency` interchangeably.
+
+== Dependency Injection Types
+
+Micronaut Framework offers three types of dependency injection:
+
+=== Constructor Injection
+
+In constructor-based injection, Micronaut Framework provides the dependencies required for the class as arguments to the constructor.
+
+source:constructor/MessageController[]
```

Comment:
Worth describing that you can use `@Nullable` to make a constructor parameter optional

Reply:
Worth describing that beyond types you can injection lists, arrays etc.

Reply:
I think we could talk about optional dependencies and multiple injection in a separate guide. I would like to keep this guide as simple as possible and focus on the different types of injection supported.